### ISSUE TITLE
Reset all control points is broken

### ISSUE THREAD

**Issue by lovebug356** [likec4/likec4#1323]
Reset all control points is broken, and produces and error.

This can be reproduced in the online playground:
https://playground.likec4.dev/w/dynamic/index/

<img width="1003" alt="image" src="https://github.com/user-attachments/assets/754b17b4-d20e-486f-8bc0-1e6d41afb3cf">

If you click the button "Reset all control points" you get the following error:
<img width="710" alt="image" src="https://github.com/user-attachments/assets/fa79d7bc-5de2-4d45-ac45-695139a46fbf">

**Comment by arthurbellis** [comment]
(Hello from Tokyo -- I've found LIkeC4 useful and was looking for an excuse to dig into the details a bit; this bug seemed like a good starter. cc: @davydkov @pavelpykhtin in case this is helpful)

I was able to reproduce the error with this minimal input & clicking "Reset all control points":

```
specification {
  element component
}

model {
  component a 'A'
}

views {
  dynamic view test_self_reference {
    include a
    a -> a "self-reference"
  }
}
```

It looks like this is due to `resetEdgeControlPoints()` setting a single control point to be along the path between source and target (via `getControlPointForEdge()`). If the source and target are the same, the control point is the same as the source (and target).

When the edge label position is updated in [`RelationshipEdge.tsx`](https://github.com/likec4/likec4/blob/e73624cb7449088d5d859811faba0f6f1b30c75b/packages/diagram/src/xyflow/edges/RelationshipEdge.tsx#L242), the path for this self-referencing edge is now empty (source, target and control points are all the same), causing the error: "Failed to execute 'getPointAtLength' on 'SVGGeometryElement': The element's path is empty."

I was able to fix it with the following change to `diagramStore.ts`:

```typescript
          resetEdgeControlPoints: () => {
            const { xyflow, scheduleSaveManualLayout, xynodes, xyedges } = get()

            xyedges
              .filter(edge => edge.source !== edge.target) // don't update self-referencing edges
              .forEach(edge => {
                xyflow.updateEdgeData(edge.id, {
                  controlPoints: getControlPointForEdge(edge)
                })
              })
[...]
```

Not sure if there is a better way to handle this, but I'm happy to send a pull request if the above looks okay.
